styx-scheduler-service/src/main/java/com/spotify/styx/state/StateUtil.java

```diff
@@ -110,21 +111,24 @@ public static Set<String> workflowResources(boolean globalConcurrenyEnabled,
   }
 
   public static boolean hasTimedOut(Optional<Workflow> workflowOpt, RunState runState, Instant instant,
-                                     Duration timeout) {
+                                     Duration defaultTimeout, Duration maxRunningTimeout) {
     if (runState.state().isTerminal()) {
       return false;
     }
 
-    final Duration effectiveTimeout = runState.state() == RunState.State.RUNNING
+    Duration effectiveTimeout = runState.state() == RunState.State.RUNNING
                                       ? workflowOpt
                                           .flatMap(workflow -> workflow.configuration().runningTimeout())
-                                          .orElse(timeout)
-                                      : timeout;
-    final Duration sanitizedTimeout = effectiveTimeout.compareTo(timeout) < 0 ? effectiveTimeout : timeout;
+                                          .orElse(defaultTimeout)
+                                      : defaultTimeout;
+
+    if (runState.state() == RunState.State.RUNNING) {
```

Comment:
Why is this `if` needed?

Reply:
isn't it needed since the function checks hasTimedOut for each runstate and there is no max value for the others?

Reply:
before for non running state it sanitized timeout wouldn't do anything as we compare timeout.compareTo(timeout)

Reply:
Something like:

```java
  public static boolean hasTimedOut(Optional<Workflow> workflowOpt, RunState runState, Instant instant,
                                    Duration runStateTimeout, Duration maxRunningTimeout) {
    if (runState.state().isTerminal()) {
      return false;
    }
    
    Duration effectiveTimeout = runStateTimeout;
    if (runState.state() == RunState.State.RUNNING) {
      effectiveTimeout = workflowOpt
          .flatMap(workflow -> workflow.configuration().runningTimeout())
          .orElse(runStateTimeout);
      effectiveTimeout = effectiveTimeout.compareTo(maxRunningTimeout) < 0 ? effectiveTimeout : maxRunningTimeout;
    }

    final Instant deadline = Instant
        .ofEpochMilli(runState.timestamp())
        .plus(effectiveTimeout);

    return !deadline.isAfter(instant);
  }
```